views/js/component/proctorForm.js

```diff
@@ -24,54 +24,117 @@ define([
+     * @param {type} $container
+     * @returns {undefined}
+     */
     function init($container){
-        
+
         $.get(proctorFormUrl, function(formData){
-            
+
             renderFormFromData($container, formData);
-            
-            $container.on('submit', 'form', function(e){
-                
+
+            $container.on('submit' + _ns, 'form', function(e){
+
                 var $form = $(this);
                 var fields = $form.serializeArray();
-                var data = {};
+                var data = {
+                    testCenters : [] //get the testCenter from the test center datalist component
+                };
+                
+                e.preventDefault();
+                
                 _.each(fields, function(field){
-                   data[field.name] = field.value; 
+                    data[field.name] = field.value;
                 });
-                
+
                 $.post(proctorFormUrl, data, function(res){
-                    console.log(res);
                     if(res.created){
                         feedback().success(__('Proctor created'));
+                        $container.empty();
                     }else{
-                        renderFormFromData($container, formData);
+                        renderFormFromData($container, res);
                     }
                 });
-                
-                e.preventDefault();
             });
         });
     }
+    
+    /**
+     * 
+     * @param {type} config
+     * @param {JQuery} config.renderTo - the jQuery container it should be rendered to
+     * @param {Object} config.testCenterList - the test center list component
+     * @returns {proctorForm}
+     */
+    return function proctorFormFactory(config){
+
+        return component()
+            .on('destroy', function(){
+                this.getElement().off(_ns).empty();
+            })
+            .on('render', function(){
+
+                var self = this;
+                var $element = this.getElement();
 
-    return {
-        init : init
+                $.get(proctorFormUrl, function(formData){
+
+                    renderFormFromData($element, formData);
+
+                    $element.on('submit' + _ns, 'form', function(e){
+
+                        var $form = $(this);
+                        var fields = $form.serializeArray();
+                        var data = {
+                            testCenters : [] //@todo get the testCenter from the test center datalist component (config.testCenterList)
+                        };
+                        _.each(fields, function(field){
+                            data[field.name] = field.value;
+                        });
+
+                        $.post(proctorFormUrl, data, function(res){
+                            if(res.created){
+                                feedback().success(__('Proctor created'));
+                                $element.empty();
```

Comment:
I think there is something missing: we need to known when the component has done its stuff, in order to go back to the list. Maybe an event or a callback ?